metrics-operator/controllers/metrics/controller.go

```diff
@@ -125,6 +115,42 @@ func (r *KeptnMetricReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	return reconcile, err
 }
 
+func (r *KeptnMetricReconciler) updateMetric(metric *metricsapi.KeptnMetric, value string, rawValue []byte, reconcile ctrl.Result, err error) (*metricsapi.KeptnMetric, ctrl.Result) {
```

Comment:
since the metric is passed as a pointer here, we do not need to return it afterwards 
```suggestion
func (r *KeptnMetricReconciler) updateMetric(metric *metricsapi.KeptnMetric, value string, rawValue []byte, reconcile ctrl.Result, err error) ctrl.Result {
```